code/modules/projectiles/guns/special/hand_of_midas.dm

```diff
@@ -0,0 +1,130 @@
+// Hand of Midas
+
+/obj/item/gun/magic/midas_hand
+	name = "The Hand of Midas"
+	desc = "An ancient Egyptian matchlock pistol imbued with the powers of the Greek King Midas. Don't question the cultural or religious implications of this."
+	ammo_type = /obj/item/ammo_casing/magic/midas_round
+	icon_state = "midas_hand"
+	inhand_icon_state = "gun"
+	worn_icon_state = "gun"
+	lefthand_file = 'icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'
+	fire_sound = 'sound/weapons/gun/rifle/shot.ogg'
+	pinless = TRUE
+	max_charges = 1
+	can_charge = FALSE
+	item_flags = NEEDS_PERMIT
+	w_class = WEIGHT_CLASS_BULKY // Balance thing, but lets just say it
+	force = 3
+	trigger_guard = TRIGGER_GUARD_NORMAL
+	antimagic_flags = NONE
+	can_hold_up = FALSE
+
+	/// How much gold reagent we have in reserves. Affects the length of the Midas Blight debuff.
+	var/gold_reagent = 2.5 SECONDS
```

Comment:
I find the var name and doc comment a tad misleading. This is actually the duration of the debuff itself.

Reply:
I realized halfway through making it that representing time as a reagent was confusing, I just forgot to swap the name too